pwa-devdocs/src/pwa-buildpack/reference/targetables/SingleImportStatement/index.md

```diff
@@ -0,0 +1,68 @@
+---
+title: SingleImportStatement
+---
+
+<!--
+The reference doc content is generated automatically from the source code.
+To update this section, update the doc blocks in the source code
+-->
+
+{% include auto-generated/pwa-buildpack/lib/WebpackTools/targetables/SingleImportStatement.md %}
+
+## Examples
+
+Code examples for using the `SingleImportStatement` class.
+
+### Create a `SingleImportStatement` object
+
+Pass in an import statement to the constructor to create a new `SingleImportStatement` object.
+
+```js
+const useQueryImport = new SingleImportStatement("import { useQuery } from '@apollo/react-hooks'");
+```
+
+This creates an object with the following properties:
+
+```js
+{
+  statement: "import { useQuery } from '@apollo/react-hooks'",
+  binding: 'useQuery',
+  imported: 'useQuery'
+}
+```
+
+### Change the binding
+
+Use the `changeBinding()` function to rename the variable bound to the imported object.
+
+```js
+const useQueryImport = new SingleImportStatement("import { useQuery } from '@apollo/react-hooks'");
+
+const useQueryImport2 = useQueryImport.changeBinding('useQuery2');
+```
+
+This creates an object with the following properties:
+
+```js
+{
+  statement: "import { useQuery as useQuery2 } from '@apollo/react-hooks'",
+  binding: 'useQuery2',
+  imported: 'useQuery'
+}
+```
+
+### Using the `SingleImportStatement` object
+
+The `toString()` value of a `SingleImportStatement` object is the value of the `binding` property.
+Use this to reference the component's local name when adding custom code with Targetables.
+
+```jsx
+let Button = new SingleImportStatement("Button from './button'");
```

Comment:
```suggestion
let Button = new SingleImportStatement("import Button from './button'");
```

Reply:
**Fun Fact:** You can skip `import` here and the class is smart enough to infer what you mean.

See: https://github.com/magento/pwa-studio/blob/develop/packages/pwa-buildpack/lib/WebpackTools/targetables/__tests__/SingleImportStatement.spec.js#L34-L46

I'll add that note in the code.

Reply:
Good to know. Nicee.